pkg/vetter/conflictingvirtualservicehost/vet.go

```diff
@@ -392,6 +398,22 @@ func (v *VsHost) Vet() ([]*apiv1.Note, error) {
 	return notes, nil
 }
 
+// check same VS route conflict
+func sameVSconflict(rule1 routeRule, rule2 routeRule) (bool, error) {
+
+	if rule1.vsName == rule2.vsName && rule1.namespace == rule2.namespace {
```

Comment:
I'm not sure if this if statement is necessary. The name of the function and the documentation for this function state that it's for the same namespace. Therefore, it's on the burden of the caller to do this check before calling this function.

Reply:
it's true. I debated to add that or not, but thought it's not hurt to have an extra check.

Reply:
In my opinion, the check is verbose enough that it makes it harder to read without much gain since it's a private function that's descriptively named.

However, if you're dead-set on keeping it, I'd recommend checking if they're _not_ in the same virtual service first (and document what the if statement is checking), return immediately if not, and then have the main logic outside of the if statement. That would have the benefit of making it much easier to read and having that extra check, if you want to keep it in.